flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -61,7 +65,22 @@ public static Schema convert(TableSchema schema) {
     RowType root = (RowType) schemaType;
     Type converted = root.accept(new FlinkTypeToType(root));
 
-    return new Schema(converted.asStructType().fields());
+    Schema iSchema = new Schema(converted.asStructType().fields());
+    return freshIdentifierFieldIds(iSchema, schema);
+  }
+
+  private static Schema freshIdentifierFieldIds(Schema iSchema, TableSchema schema) {
```

Comment:
instead of having this method, if we expose a `TypeUtil.freshIdentifierFieldIds(Schema iSchema, Schema base)`, can we do a conversion of Flink to Iceberg schema and then call that method?

Reply:
I don't think we can reuse the `TypeUtil.refreshIdentifierFields(Types.StructType freshSchema, Schema baseSchema)` because in this method we've had an existing identifier field id list inside the `baseSchema`.  While for this case,  the identifier fields are actually came from flink's `TableSchema`,  converting the flink's `TableSchema` with primary keys to the iceberg table `schema` with identifier field id list is exactly the thing we are trying to accomplish in this method.